### DISCUSSION TITLE
importing experiment data

### DISCUSSION THREAD

**Discussion by TheRich-71** [TheELNConsortium/TheELNFileFormat#31] [Q&A]
Dear members of this consortium,

Besides my position at a university, I am running a small company producing instruments for researchers. I stumbled across this project some time ago and asked our software developer to add the possibility of enveloping the files generated in an experiment into a *.eln. By this, users should be able to easily transfer their measured raw data into an experiment of their electronic labbook.

We have been able to test this with elabFTW, which allows easily importing ELNFiles and creating new experiments with the attached files, but I have no access to the remaining electronic labbooks. In the demo instance of Kadi4mat, i.e., I did not find a way to import ELNFiles - or it did not appear obvious to me.

The question is, can we estimate that a successful test in elabFTW is proving that our export format can be imported in the remaining consortiums systems, or should we run individual tests?

**Comment by NicolasCARPi** [comment]
Hello Thomas,

> can we estimate that a successful test in elabFTW is proving that our export format can be imported in the remaining consortiums systems

While it probably means the import will mostly work, it is still worthwile to try it out everywhere. For that I recommend that you become part of the consortium (your company), and add an example in the [examples folder](https://github.com/TheELNConsortium/TheELNFileFormat/tree/master/examples) so that **we** can try it out and let you know if things need adjusting.

We will also probably at some point add a validator as a github action so obvious errors can be catched fast.

**Comment by nicobrandt** [comment]
About Kadi4Mat, while we haven't added an import in the web interface yet, you could try it with our API library if you want, which can be configured to work together with the Demo instance. It also includes a CLI command for a quick test: https://kadi-apy.readthedocs.io/en/latest/usage/cli.html#kadi-apy-misc-import-file

**Reply by adamreupert** [reply]
Thanks for the clarification. 
I wanted to test something quickly and instead of using the API libary I searched on the web interface and couldn't find it. Maybe worth pointing this out in the Kadi4mat-Example Readme.md. 
Out of curiosity is there a plan to add this format to the web interface besides the JSON import option in the future?
